docs/DeveloperGuide.md

```diff
@@ -294,6 +294,61 @@ of a `StatisticsCommand`:
     * Pros: Does not violate the law of demeter. Increases cohesion and thus increase maintainability.
     * Cons: Requires more wrapper methods to carry information. More effort to implement.
 
+### Attendance Model
+This section explains the design considerations of the `Attendance` model.
+
+#### Implementation
+The `Attendance`, `AttendanceRecord` and `AttendanceRecordList` models are implemented as follows:
+
+![AttendanceClassDiagram](images/AttendanceClassDiagram.png)
+
+`Attendance` encapsulates the information of a particular student's attendance in a specific lesson he/she has attended.
+`AttendanceRecord` refers to the actual attendance roster of a particular lesson event.
+It contains the `Attendances` of all `Students` who have attended the particular lesson.
+These information are stored as a `Map` with `Student` `UUID` as keys.
+`AttendanceRecordList` refers to the list of all `AttendanceRecord` instances.
+The size of this list is fixed and is determined by the `NumberOfOccurences` in `Lesson`.
+
+<div markdown="span" class="alert alert-primary">
+Note that all classes in the `Attendance` package are designed to be immutable.
+</div>
+
+### Design Considerations:
+#### Aspect: Reducing user input
+Users should not have to type in attendance related commands when the student is absent from a lesson.
+Therefore, the `Attendance` class does not have, for example, a `boolean hasAttended` attribute.
+It only has a `participationScore` attribute.
+
+#### Aspect: Maintaining immutability and optimising `AttendanceRecord`
+* **Alternative 1 (current choice):** Dynamically updating `AttendanceRecord` whenever there is a change to attendance.
+    * Pros: Guarantees immutability.
+    * Cons: Requires re-instantiation of a `Map` object whenever a user adds/edits/deletes an `Attendance`.
+
+* **Alternative 2:** Initialising empty `Attendance`s for all students on call to constructor method. \
+This would mean that each `Attendance` is set to a particular value, rather than having to copy the entire `Map` object, whenever a user adds/edits/deletes it.
+    * Pros: Less overhead in modifying `Attendance`.
+    * Cons:
+      * Violates immutability.
+      * Incurs greater memory use.
+
+#### Aspect: Handling of invalid `Week` number
+* **Alternative 1 (current choice):** Fixed size `List`.
+    * Pros:
```

Comment:
I'll send in a separate PR to update my pros and cons section as well. Lets follow your style of having the pros and cons as individual pointers rather than paragraphs